Prevents circular linking of docs

codingCoffee opened the pull request:
- also added test case to check if code to prevent circular linking of
doctypes works

codingCoffee committed:
Prevents circular linking of docs

- also added test case to check if code to prevent circular linking of
doctypes works

rmehta commented on frappe/core/doctype/communication/test_communication.py:
```diff
@@ -43,3 +43,29 @@ def test_name(self):
+		a = frappe.get_doc("Communication", a.name)
+		a.reference_doctype = "Communication"
+		a.reference_name = c.name
+		self.assertRaises(frappe.ValidationError, a.save)
```
throw a specific error `frappe.CircularLinkingError` and catch it

codingCoffee committed:
created a CircularLinkingError class

- this can be used later if circular linking error needs to be thrown
elsewhere
